File: DateChecker.java

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class DateChecker {
	public static boolean isEffectiveDateInRange(String effectiveDateStr, String partBEffectiveDateStr) {
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM/dd/yy");
		LocalDate partBEffectiveDate = LocalDate.parse(partBEffectiveDateStr, formatter);
		LocalDate earliestDate = partBEffectiveDate.minusMonths(3).withDayOfMonth(1);
		System.out.println(earliestDate);
		LocalDate lastDate = partBEffectiveDate.plusMonths(5)
				.withDayOfMonth(partBEffectiveDate.plusMonths(5).lengthOfMonth());
		System.out.println(lastDate);
		LocalDate effectiveDate = LocalDate.parse(effectiveDateStr, formatter);
		return effectiveDate.isAfter(earliestDate) && effectiveDate.isBefore(lastDate)
				|| effectiveDate.isEqual(earliestDate) || effectiveDate.isEqual(lastDate);
	}
	
	public static void main(String[] args) {
		
		System.out.println(isEffectiveDateInRange("01/01/23","05/02/23"));
		
	}
}

File: DateChecker.js

function isEffectiveDateInRange(effectiveDateStr, partBEffectiveDateStr) {
    const formatter = new Intl.DateTimeFormat('en-US', {month: '2-digit', day: '2-digit', year: '2-digit'});
    var partBEffectiveDate = new Date(formatter.format(new Date(partBEffectiveDateStr)));
    const earliestDate = new Date(partBEffectiveDate.setMonth(partBEffectiveDate.getMonth() - 3));
    earliestDate.setDate(1);
     console.log(earliestDate);
     partBEffectiveDate = new Date(formatter.format(new Date(partBEffectiveDateStr))); 
    const lastDate = new Date(partBEffectiveDate.setMonth(partBEffectiveDate.getMonth() + 5));
    lastDate.setDate(0);
    
    const effectiveDate = new Date(formatter.format(new Date(effectiveDateStr)));
    console.log(lastDate);
    return effectiveDate > earliestDate && effectiveDate < lastDate || effectiveDate.getTime() === earliestDate.getTime() || effectiveDate.getTime() === lastDate.getTime();
  }
  console.log(isEffectiveDateInRange("09/01/23", "04/02/23"));
  